Dispatcher/Views/CustomHeaderView.swift

```diff
@@ -0,0 +1,102 @@
+import UIKit
+
+protocol CustomHeaderViewDelegate: AnyObject {
+    func firstRightIconPressed()
+    func secondRightIconPressed()
+    func leftIconPressed()
+}
+
+extension CustomHeaderViewDelegate {
+    func firstRightIconPressed() {}
+    func secondRightIconPressed() {}
+    func leftIconPressed() {}
+}
+
+
+class CustomHeaderView: UIView {
+    
+    @IBOutlet weak var firstRightImageView: UIImageView!
+    @IBOutlet weak var secondRightImageView: UIImageView!
+    @IBOutlet weak var leftImageView: UIImageView!
+    @IBOutlet var contentView: UIView!
+    
+    private weak var delegate: CustomHeaderViewDelegate?
+
+    
+    func initView(delegate: CustomHeaderViewDelegate? = nil, icon1: UIImage? = nil, icon2: UIImage? = nil, leftIcon: UIImage? = nil) {
+        
+        commonInit()
```

Comment:
You are calling commonInit() and define the tap UITapGestureRecognizer for **all** the buttons. Only after, you check if you should show the button or not. Not so great.

Reply:
Fixed on the updated version (next PR)
@abenartz